torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -531,45 +528,25 @@ class CppMicroGemmAMX(CppMicroGemm):
     alignas(4096) {{input_t}} dequantized_B_buf[buf_size];
     {%- endif %}
 
-    const auto b_tile_ptr_stride = ldb * {{vnni_size}};
-
-    auto load_B_row = [&]({{input2_t}}* {{restrict_keyword}} src, {{input_t}}* {{restrict_keyword}} dst) {
-        auto b_int8 = at::vec::Vectorized<int8_t>::loadu(src, static_cast<int64_t>(32));
-        auto b_bf16 = at::vec::convert<{{input_t}}>(b_int8);
-        b_bf16.store(dst);
-    };
-
-    auto load_B_tile = [&]({{input2_t}}* B_ptr, int idx, int num_b_rows) {
-        {{input_t}}* base_addr = dequantized_B_buf + idx;
-        {{kernel.unroll_pragma(8)}}
-        for (int i = 0; i < num_b_rows; i++) {
-            load_B_row(
-                B_ptr + i * b_tile_ptr_stride,
-                base_addr + i * 32
-            );
-        }
-    };
-    auto load_dequantized_B = [&](int n) {
+    auto load_dequantized_B = [&](int tile_idx) {
         // Load a tile of B & cache it in L1D.
-        {{kernel.unroll_pragma(4)}}
-        for (int k = 0; k < K; k += {{block_k}}) {
-            int num_b_rows = (k < last_k_offset) ? 16 : tail_k_size;
-            {{kernel.unroll_pragma(2)}}
-            for (int tile_col = 0; tile_col <= 1; tile_col++) {
-                load_B_tile(
```

Comment:
Hi @leslie-fang-intel, the previous approach placed the elements of an AMX B tile contiguously.
I'm surprised to see that the new approach doesn't result in a performance penalty. 😮

Reply:
TBH, I haven't checked the performance of patch https://github.com/leslie-fang-intel/pytorch/commit/4c83e4e75138e8fa6e0d58438f75b7718dc8a0cc#diff-139642bd981df977f70f4c18c1c34bd1a85c1d6b9ffa06aaa98426ed83942a31R537